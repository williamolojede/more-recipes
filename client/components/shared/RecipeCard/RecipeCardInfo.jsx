import React from 'react';

import { recipeCardInfoPropTypes } from '../../../config/proptypes';
import defaultConfig from '../../../config/default';

import RecipeStats from '../RecipeStats.jsx';


const RecipeCardInfo = (props) => {
  const { User, upVoteCount, downVoteCount, favoriteCount, viewCount } = props.info;

  return (
    <div className="card-action recipe-card__info">
      {/* if user img is present show it or the default if not */}
      <img
        src={User.imgUrl || defaultConfig.userImgUrl}
        alt={User.fullname}
        title={User.fullname}
        className="recipe-card__info---img"
      />
      <div className="recipe-card__info">
        <span><span className="recipe-card__info-by">Recipe by</span> {User.fullname}</span>
        <RecipeStats />
        <ul className="recipe-card__info--stats">
          <li className="stats__favorite">
            <i className="mdi mdi-heart" />
            <span>{favoriteCount}</span>
          </li>
          <li className="stats__vote-up">
            <i className="mdi mdi-thumb-up" />
            <span>{upVoteCount}</span>
          </li>
          <li className="stats__vote-down">
            <i className="mdi mdi-thumb-down" />
            <span>{downVoteCount}</span>
          </li>
          <li className="stats__view">
            <i className="mdi mdi-eye" />
            <span>{viewCount}</span>
          </li>
        </ul>
      </div>
    </div>
  );
};

RecipeCardInfo.propTypes = recipeCardInfoPropTypes;
export default RecipeCardInfo;
