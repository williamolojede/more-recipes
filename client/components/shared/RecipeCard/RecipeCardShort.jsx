import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const RecipeCardShort = ({
  id,
  name,
  removeRecipe,
  type
}) => (
  <li className="col s12 m6 l4 x4 recipes-list-item">
    <div className="card">
      <Link to={`/recipe/${id}`} className="recipes-list-item__name">
        {name}
      </Link>
      <div className="recipes-list-item__actions">
        {
          type !== 'favorite' &&
          <Link
            to={`/user/recipes/${id}/edit`}
            className="recipes-list-item__actions-edit"
          >
            <i className="mdi mdi-pencil" />
          </Link>
        }
        <button
          onClick={() => removeRecipe(id)}
          className="recipes-list-item__actions-delete"
        >
          <i className="mdi mdi-delete" />
        </button>
      </div>
    </div>
  </li>
);

RecipeCardShort.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  removeRecipe: PropTypes.func.isRequired,
  type: PropTypes.oneOf(['favorite', 'personal']).isRequired
};

export default RecipeCardShort;
