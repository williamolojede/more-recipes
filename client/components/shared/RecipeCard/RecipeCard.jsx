import React from 'react';
import { Link } from 'react-router-dom';
import RecipeCardInfo from './RecipeCardInfo.jsx';

const RecipeCard = () => (
  <div className="recipe-card card">
    <Link to="/repcipe/1">
      <img
        src="https://fthmb.tqn.com/4P30MYPyJYr0fxlkBuHmNZg4NO8=/960x0/filters:no_upscale()/about/3167488129_cc7a303896_o-58a6fe3b5f9b58a3c91bf559.jpg"
        alt="pancake"
        className="recipe-card__img card-image"
      />
    </Link>
    <div className="card-content">
      <Link to="/repcipe/1">
        <h4 className="recipe-card__title card-title">Pancake</h4>
      </Link>
      <p className="recipe-card__description">Who doesnâ€™t love pancakes in the morning? These delicious vanilla pancakes are thick and </p>
    </div>
    <RecipeCardInfo />
  </div>
);

export default RecipeCard;
