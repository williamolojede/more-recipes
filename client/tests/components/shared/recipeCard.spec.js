import React from 'react';
import RecipeCard from '../../../components/shared/RecipeCard/RecipeCard';
import recipesMock from '../../__mocks__/recipes.mock';

function setup() {
  const props = {
    recipe: recipesMock.recipes[0],
  };
  const wrapper = shallow(<RecipeCard {...props} />);

  return {
    wrapper
  };
}

describe('Recipe Card', () => {
  it('should render with right amount of elements', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
