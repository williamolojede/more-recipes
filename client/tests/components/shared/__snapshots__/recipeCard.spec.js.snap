// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe Card should render with right amount of elements 1`] = `
<div
  className="recipe-card card"
>
  <Link
    className="recipe-card__img"
    replace={false}
    to="/recipe/3"
  >
    <img
      alt="pancake"
      className="card-image"
      src="https://firebasestorage.googleapis.com/v0/b/more-recipes.appspot.com/o/recipes%2F1509702436404?alt=media&token=da3a7ac8-58d3-46da-a71d-c2c1c0cbdfeb"
    />
  </Link>
  <div
    className="card-content"
  >
    <Link
      replace={false}
      to="/recipe/3"
    >
      <h4
        className="recipe-card__title card-title"
      >
        Chef John's Spaghetti with Red Clam Sauce
      </h4>
    </Link>
    <p
      className="recipe-card__description"
    >
      This is one of my 'go to' recipes and it takes only about 15 minutes from start to finish. The only prep work really is mashing the garlic. The sauce is great on any pasta, but I prefer angel hair or spaghetti.
    </p>
  </div>
  <RecipeCardInfo
    info={
      Object {
        "User": Object {
          "fullname": "John Bull",
          "id": 8,
          "username": null,
        },
        "downVoteCount": 0,
        "favoriteCount": 0,
        "upVoteCount": 1,
        "viewCount": 1,
      }
    }
  />
</div>
`;
